src/mpi/events.rs

```diff
@@ -57,6 +65,17 @@ impl EventsPublisher {
             handler.finish();
         }
     }
+
+    pub fn get_travel_time_collector(&mut self) -> Option<&TravelTimeCollector> {
```

Comment:
This is something we want, I don't think this should be in here though. Either this should be something like

```
pub fn get_subscriber<T: EventsSubscriber>(&self) -> Option<&T>
```
Or we make the `subscribers` field accessible. The generic version looks nice, what if we have two subscribers of the same type though.

Reply:
Good point, I made it generic. Let's talk about the case if we have more than one subscriber.